fastapi/security/http.py

```diff
@@ -95,9 +93,8 @@ def __init__(
         scheme_name: str = None,
         auto_error: bool = True,
     ):
+        super().__init__(scheme="", scheme_name=scheme_name, auto_error=auto_error)
```

Comment:
I totally agree with you.

I think we should respect base classes and invoke their `__init__` methods, that why I did so.
In case if it wrong I will revert to the base version. 

@SirTelemak what is your opinion?

Reply:
Unlike previous point, this one is okay with `super()` just wanted to draw you attention here

Reply:
I think the best option here is to extract the same logic of `HTTPBearer` and `HTTPBase` to base class.
Looks like `HTTPBearer` doesn't match to the logic of a base `HTTPBase` class.